internal/config/generate/main.go

```diff
@@ -17,24 +17,197 @@ import (
 	"github.com/BurntSushi/toml"
 )
 
+// This script will read the Config.toml file and create:
+// - a formatted get.go file, with get functions for each environment variable;
+// - a config.md file with documentation for the environment variables.
+//
+// Each table entry in the toml file translates into an environment variable.
+// In Go, this becomes a map[string]Env, with the keys of the map being the variables' names.
+func main() {
+	data := readTOML("generate/Config.toml")
+	config := decodeTOML(data)
+	sortedTopics, sortedEnvs := sortConfig(config)
+
+	var code strings.Builder
+	var doc strings.Builder
+
+	addCodeHeader(&code)
+	addDocHeader(&doc)
+
+	for _, topic := range sortedTopics {
+		for _, env := range sortedEnvs[topic] {
+			e := config[topic][env]
+			e.validate()
+			addLine(&code, e.toFunction())
+			addLine(&doc, e.toDoc())
+		}
+	}
+
+	writeToFile("get.go", formatCode(code.String()))
+	writeToFile("config.md", []byte(doc.String()))
+}
+
+// ------------------------------------------------------------------------------------------------
+// Auxiliary functions
+// ------------------------------------------------------------------------------------------------
+
+func readTOML(name string) string {
+	bytes, err := os.ReadFile(name)
+	if err != nil {
+		panic(err)
+	}
+	return string(bytes)
+}
+
+type configTOML = map[string](map[string]*Env)
+
+func decodeTOML(data string) configTOML {
+	var config configTOML
+	_, err := toml.Decode(data, &config)
+	if err != nil {
+		panic(err)
+	}
+	return config
+}
+
+// Creates sorted lists from the config to make the generated files deterministic.
+// Returns a (sorted) list of topics
+// and a map of topics to (sorted) lists of environment variable names.
+func sortConfig(config map[string](map[string]*Env)) ([]string, map[string][]string) {
+	var topicNames []string
+	topicEnvs := make(map[string]([]string))
+	for topicName, topic := range config {
+		topicNames = append(topicNames, topicName)
+		var envNames []string
+		for envName, env := range topic {
+			env.Name = envName // initializes the environment variable's name
+			envNames = append(envNames, envName)
+		}
+		sort.Strings(envNames)
+		topicEnvs[topicName] = envNames
+	}
+	sort.Strings(topicNames)
+
+	return topicNames, topicEnvs
+}
+
+func addLine(builder *strings.Builder, s string, a ...any) {
+	builder.WriteString(fmt.Sprintf(s, a...))
+	builder.WriteString("\n")
+}
+
+func addCodeHeader(builder *strings.Builder) {
```

Comment:
Could you move functions related to the printing to two separate files? One for code printing and another one for docs printing.